📖Update `amp-access` and `amp-subscriptions` docs (#27672)
ampproject/amphtml: extensions/amp-subscriptions/amp-subscriptions.md

# amp-subscriptions

## Usage

The `amp-subscriptions` extension implements subscription-style access/paywall rules.

> The solution comprises the following components:

Reviewer: Could you elaborate on why this section is here? What do you mean by "components"? Because AMP is built using AMP's components this is a tricky word.

Author: It's meant to give a high-level overview of the various parts that make up the component. Maybe this is redundant with the "Page landing flow" section. WDYT?

   2. [**Login page**][4]: provided by the publisher, allows the Publisher to authenticate the Reader and connect their identity with AMP Reader ID.
   3. [**Subscribe page**][5]: provided by the publisher, allows the Reader to purchase a subscription from the Publisher.
   4. [**Pingback endpoint**][6]: provided by the Publisher, is used to send the “view” impression for a document.
3. [**Vendor services**][7]: registered AMP extensions that cooperate with the main `amp-subscriptions` extension.
4. [**Fallback entitlement**][8]: provided by the Publisher, determines what the Reader can see in the event that all services fail to respond to the calls made to the Authorization endpoints.
5. [**Service score factors**][9]: provided by the Publisher, used if no service returns a valid entitlement to determine which service is used as the default service.
6. [**Structured data markup**][10]: Schema.org page-level configuration, provided by the Publisher.
7. [**Attributes**][11]: authored by the Publisher, defines which parts of a document are visible in which circumstances.
8. [**Actions**][12]: fulfilled by a particular service to present the Reader with a specific experience.

> ### Flow

Reviewer: This heading is vague to me - what flow? Set up flow? The flow the user follows?

Author: This describes how the parts interact when a user opens an article that contains amp-subscriptions. Open to suggestions here on the wording.

Author: SGTM!

> 1. Google AMP Cache returns the document to the Reader with some sections obscured using [Attributes][11].

Reviewer: What is the Reader? Is there supplementary documentation we could link to? How is is related to AMP Reader ID?

Author: This was detailed in the glossary that I copied over from the amp-access page: https://amp.dev/documentation/components/amp-access/#amp-glossary.

> 1. Google AMP Cache returns the document to the Reader with some sections obscured using [Attributes][11].

Reviewer: Does this flow only work for Google's AMP cache?

Author: Actually, no - it doesn't necessarily come from the AMP cache. Correct me if I'm wrong here @jpettitt.

Author: Yep. Changing this to:
> The AMP Page is loaded in the AMP viewer with some sections obscured using [Attributes][11].

support it is recommended that new publisher and paywall provider implementations
use `amp-subscriptions`.

### AMP Reader ID

To assist access services and use cases, AMP Access introduced the concept of _Reader ID_.

> The Reader ID is an anonymous and unique ID created by the AMP ecosystem. It is unique for each Reader/Publisher pair - a Reader is identified differently to two different Publishers. It is a non-reversible ID. The Reader ID is included in all AMP/Publisher communications and has very high entropy. Publishers can use the Reader ID to identify the Reader and map it to their own identity systems.

Reviewer: This sentence is confusing to me:
`a Reader is identified differently to two different Publishers.`

The same reader is given a different id? Or a reader with one id is identified separately by publishers?

Reviewer: I do not understand this sentence at all:
`The Reader ID is included in all AMP/Publisher communications and has very high entropy.`

What do you mean by high entropy?

Author: >a Reader is identified differently to two different Publishers.

This means that the same reader is represented by different ids to different publishers.

>The Reader ID is included in all AMP/Publisher communications and has very high entropy.

My understanding here is that the reader ID is very obscure to the publisher so cannot be used to try and determine any facts about the user or where they are browsing from. @jpettitt - is that right?

> The Reader ID is an anonymous and unique ID created by the AMP ecosystem. It is unique for each Reader/Publisher pair - a Reader is identified differently to two different Publishers. It is a non-reversible ID. The Reader ID is included in all AMP/Publisher communications and has very high entropy. Publishers can use the Reader ID to identify the Reader and map it to their own identity systems.

Author: I've added that paragraph in.

After reading [this](https://en.wikipedia.org/wiki/Entropy_(computing)), I think it makes sense in the context but perhaps we could say something like:

> The Reader ID is an anonymous and unique ID created by the AMP ecosystem. It is unique for each Reader/Publisher pair - a Reader is identified differently to two different Publishers. It is a non-reversible ID. It is intended to be random in nature and uses a number of factors to achieve that unpredictability. The Reader ID is included in all AMP/Publisher communications and can be used by Publishers to identify the Reader and map it to their own identity systems.

The Reader ID is constructed on the user device and intended to be long-lived. However, it follows the normal browser storage rules, including those for incognito windows. The intended lifecycle of a Reader ID is 1 year between uses or until the user clears their cookies. The Reader IDs are not currently shared between devices.

The Reader ID is constructed similarly to the mechanism used to build ExternalCID described [here](https://docs.google.com/document/d/1f7z3X2GM_ASb3ZCI_7tngglxwS6WoWi1EB3aKzdf6vo/edit#heading=h.hb9q0wpwwhuf). An example Reader ID is `amp-OFsqR4pPKynymPyMmplPNMvxSTsNQob3TnK-oE3nwVT0clORaZ1rkeEz8xej-vV6`.

### Configuration

> The `amp-subscriptions` extension must be configured using JSON configuration:

Reviewer: Please update `extension` to `component`

> The `amp-subscriptions` extension must be configured using JSON configuration:

Reviewer: ```suggestion
The `amp-subscriptions` component must contain the following fields in its JSON configuration:
```

<table>
  <tr>
    <th>Property</th>
    <th>Values</th>
    <th>Description</th>
  </tr>
  <tr>
    <td class="col-fourty"><code>services</code></td>
    <td>&lt;array&gt; of &lt;object&gt;</td>
    <td>This <code>array</code> must include:<ul><li>One <a href="#local-service">Local Service</a></li><li>Zero or more <a href="#vendor-services">Vendor Services</a>.</li></ul></td>
  </tr>
  <tr>
    <td class="col-fourty"><code>score</code></td>
    <td>&lt;object&gt;</td>
    <td>Determines which service is selected if no valid entitlements are returned.<br/>See <a href="#service-score-factors">Service Score Factors</a> for more details.</td>
  </tr>
  <tr>
    <td class="col-fourty"><code>fallbackEntitlement</code></td>
    <td>&lt;object&gt;</td>
    <td>Determines what level of access the Reader should have if all services fail to respond to the Authorization requests.<br/>See <a href="#fallback-entitlement">Fallback Entitlement</a> for more details.</td>
  </tr>
</table>

> Here is an example of a configuration:

Reviewer: ```suggestion
Below is an example of a configuration:
```

```html

> <script type="application/json" id="amp-subscriptions">

Reviewer: does the id field need to point to `amp-subscriptions` or the id of the the amp-subscriptions component?

Author: It needs to point to `amp-subscriptions` as this is how the runtime knows which specific configuration is for `amp-subscriptions` vs. other components.

  {
    "services": [
      {
        // Local service (required)
        "authorizationUrl": "https://pub.com/amp-authorisation?rid=READER_ID&url=SOURCE_URL",
        "pingbackUrl": "https://pub.com/amp-pingback?rid=READER_ID&url=SOURCE_URL",
        "actions":{
          "login": "https://pub.com/amp-login?rid=READER_ID&url=SOURCE_URL",
          "subscribe": "https://pub.com/amp-subscribe?rid=READER_ID&url=SOURCE_URL"
        }
      },
      {
        // Vendor services (optional)
        "serviceId": "service.vendor.com"
      }
    ],
    "score": {
      "supportsViewer": 10,
      "isReadyToPay": 9
    },
    "fallbackEntitlement": {
      "source": "fallback",
      "granted": true,
      "grantReason": "SUBSCRIBER",
      "data": {
        "isLoggedIn": false
      }
    }
  }
</script>
```

### Local service

> The following properties are defined in this configuration:

Reviewer: What configuration? The same one as above?

Reviewer: What is a local service vs a local configuration? If the configuration example below is for a local service what is the above configuration for?

Author: This is diving into the "local service" part of the overall configuration.  I will add some explanatory text at the beginning of the section.

Author: Changed to:
>The local service is provided by the Publisher to control and monitor access to documents.
>
>It is configured using the following properties:

  </tr>
  <tr>
    <td class="col-fourty"><code>actions.login</code></td>
    <td>&lt;URL&gt;</td>
    <td>The HTTPS URL for the <a href="#login-page">Login page</a>.</td>
  </tr>
  <tr>
    <td class="col-fourty"><code>actions.subscribe</code></td>
    <td>&lt;URL&gt;</td>
    <td>The HTTPS URL for the <a href="#subscribe-page">Subscribe page</a>.</td>
  </tr>
</table>

_&lt;URL&gt;_ values specify HTTPS URLs with substitution variables. The substitution variables are covered in more detail in the [URL Variables][13] section below.

Here’s an example of a "local" service configuration:

```html
<script type="application/json" id="amp-subscriptions">
  {
    "services": [
      {
        "authorizationUrl": "https://pub.com/amp-authorisation?rid=READER_ID&url=SOURCE_URL",
        "pingbackUrl": "https://pub.com/amp-pingback?rid=READER_ID&url=SOURCE_URL",
        "pingbackAllEntitlements": true,
        "actions":{
          "login": "https://pub.com/amp-login?rid=READER_ID&url=SOURCE_URL",
          "subscribe": "https://pub.com/amp-subscribe?rid=READER_ID&url=SOURCE_URL"
        }
      },
      ...
    ]
  }
</script>
```

#### URL variables

> When configuring the URLs for various endpoints, the Publisher can use substitution variables. The full list of these variables are defined in the [AMP Var Spec](https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md). In addition, this spec adds a few subscriptions-specific variables such as `READER_ID` and `AUTHDATA`.

Reviewer: ```suggestion
When configuring the URLs for various endpoints, the Publisher can use substitution variables. The full list of these variables are defined in [AMP HTML URL Variable Substitutions](https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md). In addition, this spec adds a few subscriptions-specific variables such as `READER_ID` and `AUTHDATA`.
```

Reviewer: `this spec adds a few subscriptions-specific` - what spec? The AMP vars spec? Or is this referring to the configuration?

Author: The AMP vars spec. Will clarify.

> Some of the most relevant variables are described in the table below:

Reviewer: Some of the most? Where are the rest? How many are there total?

Author: Will re-word this to be more explicit.

  </tr>
  <tr>
    <td class="col-thirty"><code>AUTHDATA(field)</code></td>
    <td>The value of the field in the authorization response.</td>
  </tr>
  <tr>
    <td class="col-thirty"><code>RETURN_URL</code></td>
    <td>The placeholder for the return URL specified by the AMP runtime for a Login Dialog to return to.</td>
  </tr>
  <tr>
    <td class="col-thirty"><code>SOURCE_URL</code></td>
    <td>The Source URL of this AMP document. If the document is served from a CDN, the AMPDOC_URL will be a CDN URL, while SOURCE_URL will be the original source URL.</td>
  </tr>
  <tr>
    <td class="col-thirty"><code>AMPDOC_URL</code></td>
    <td>The URL of this AMP document.</td>
  </tr>
  <tr>
    <td class="col-thirty"><code>CANONICAL_URL</code></td>
    <td>The canonical URL of this AMP document.</td>
  </tr>
  <tr>
    <td class="col-thirty"><code>DOCUMENT_REFERRER</code></td>
    <td>The Referrer URL.</td>
  </tr>
  <tr>
    <td class="col-thirty"><code>VIEWER</code></td>
    <td>The URL of the AMP Viewer.</td>
  </tr>
  <tr>
    <td class="col-thirty"><code>RANDOM</code></td>
    <td>A random number. Helpful to avoid browser caching.</td>
  </tr>
</table>

> Here’s an example of the URL extended with Reader ID, Canonical URL, Referrer information and random cachebuster:

Reviewer: ```suggestion
Below is an example of the URL extended with Reader ID, Canonical URL, Referrer information and random cache-buster:
```

```http
https://pub.com/amp-authorization?
   rid=READER_ID
  &url=CANONICAL_URL
  &ref=DOCUMENT_REFERRER
  &_=RANDOM
```

`AUTHDATA` variable is available to Pingback and Login URLs. It allows passing any field in the authorization
response as an URL parameter. E.g. `AUTHDATA(data.isLoggedIn)`.

#### Authorization endpoint

Authorization is an endpoint provided by the Publisher and called by the AMP Runtime. It is a credentialed CORS GET endpoint.

> **Note:** The Authorization endpoint must implement the security protocol described in the

Reviewer: ```suggestion
The Authorization endpoint must implement the security protocol described in
```

> [AMP CORS Security Spec](https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cors-requests#cors-security-in-amp).

Reviewer: ```suggestion
[CORS security in AMP](https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cors-requests#cors-security-in-amp).
```

> This endpoint returns the Entitlements object that can be used by the [Attributes][11] to hide or show different parts of the document. Authorization endpoint is specified using the "authorizationUrl" property in the config.

Reviewer: ```suggestion
The authorization endpoint returns the Entitlements object that can be used by the [Attributes][11] to hide or show different parts of the document. Authorization endpoint is specified using the "authorizationUrl" property in the config.
```

> The Entitlement response returned by the authorization endpoint must conform to the predefined format:

Reviewer: What format? Below is a table.

Author: Will change "format" to "specification, as defined in the table below"

<table>
  <tr>
    <th>Property</th>
    <th>Values</th>
    <th>Description</th>
  </tr>
  <tr>
    <td class="col-fourty"><code>granted</code></td>
    <td>&lt;boolean&gt;</td>
    <td>Stating whether or not the Reader has access to the document or not.</td>
  </tr>
  <tr>
    <td class="col-fourty"><code>grantReason</code></td>
    <td>&lt;string&gt;</td>
    <td>The reason for giving the access to the document, recognized reasons are: <ul><li><code>"SUBSCRIBER"</code> meaning the user is fully subscribed.</li><li><code>"METERING"</code> meaning user is on metering.</li></ul></td>
  </tr>
  <tr>
    <td class="col-fourty"><code>data</code></td>
    <td>&lt;object&gt;</td>
    <td>Free-form data which can be used for template rendering, e.g. messaging related to metering or article count. See <a href="#customising-content">Customising Content</a> for more details.</td>
  </tr>
</table>

> Example response for a Reader who is a subscriber and is logged into their account:

Reviewer: Wait - are Reader's users?

Author: Yep 😄

Reviewer: lol, thanks

```js
{
  "granted": true,
  "grantReason": "SUBSCRIBER",
  "data" : {
    "isLoggedIn": true
  }
}
```

Example response for an anonymous Reader who has read 4 out of 5 free articles:

```js
{
  "granted": true,
  "grantReason": "METERING",
  "data" : {
    "isLoggedIn": false,
    "articlesRead": 4,
    "articlesLeft": 1,
    "articleLimit": 5
  }
}
```

Example response for an anonymous Reader who does not have access because they have read 5 out of 5 free articles:

```js
{
  "granted": false,
  "data" : {
    "isLoggedIn": false,
    "articlesRead": 5,
    "articlesLeft": 0,
    "articleLimit": 5
  }
}
```

> Notice, while it's not explicitly visible, all vendor services also implement authorization endpoints of their own and conform to the same response format.

Reviewer: Where would I have noticed this? Is it pointed out somewhere?

Author: It's implicit from the description in the "Flow" section where we say:

> The AMP Runtime calls the Authorization endpoint of all configured services.

Reviewer: Maybe reword to be more clear:
`All vendor services must implement authorization endpoint of their own and conform to the amp-subscriptions [response format](link to response format here please).`

#### Login page

> This flow will be triggered as a result of a `"login"` action as described in the [Actions][12] section.

Reviewer: I don't understand what we mean by flow?

Author: Added some more detail here.

> An example of a Login page URL:

Reviewer: Please update all example intro to something along the lines of "Below is an example of X"

Author: Sure!

```json
{
  "actions": {
    "login": "https://pub.com/amp-login?rid=READER_ID&url=SOURCE_URL",
    ...
  }
}
```

The URL can take any parameters as defined in the [URL Variables][13] section.

#### Subscribe page

> This flow will be triggered as a result of a `"subscribe"` action as described in the [Actions][12] section.

Reviewer: Same issue with flow as above.

Author: Added some more detail here.

An example of a Subscribe URL:

```json
{
  "actions": {
    "subscribe": "https://pub.com/amp-subscribe?rid=READER_ID&url=SOURCE_URL",
    ...
  }
}
```

The URL can take any parameters as defined in the [URL Variables][13] section.

#### Pingback endpoint

Pingback is an endpoint provided by in the "local" service configuration and called by the AMP Runtime. It is a credentialed CORS POST endpoint.

**Note:** The Pingback endpoint must implement the security protocol described in the

> [AMP CORS Security Spec](https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cors-requests#cors-security-in-amp).

Reviewer: ```suggestion
[CORS security in AMP](https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cors-requests#cors-security-in-amp).
```

See the vendor service's documentation for details.

#### Available vendor services

- [amp-subscriptions-google](../amp-subscriptions-google/amp-subscriptions-google.md)

### Service score factors

> If no service returns an entitlement that grants access, all services are compared by calculating a score for each and the highest scoring service is selected. Each service has a `"baseScore"` (default 0). A value < 100 in the `baseScore` key in any service configuration represents the initial score for that service. If no `baseScore` is specified it defaults to `0`.

Reviewer: What is meant by "service"?

Author: The service refers to either a local service or one of the vendor services configured in the component. Do you think we should call that out in this section or perhaps in a dedicated section somewhere?

Reviewer: Got it. I'll take another look at it when some of the other details above are filled in. I think it ight be okay. The sentence might just need to get broken up :)

**Note:** If you would like to test the behavior of a document in the context of a particular viewer, you can add `#viewerUrl=` fragment parameter. For instance, `#viewerUrl=https://www.google.com` would emulate the behavior of a document inside a Google viewer.

### Fallback entitlement

If all configured services fail to get the entitlements, the entitlement configured under `fallbackEntitlement` section will be used as a fallback entitlement for `local` service. The document is unblocked based on this fallback entitlement.

Example fallback entitlement:
```js
{
  "fallbackEntitlement": {
    "source": "fallback",
    "granted": true,
    "grantReason": "SUBSCRIBER",
    "data": {
      "isLoggedIn": false
    }
  }
}
```

### Structured data markup

`amp-subscriptions` relies on the Schema.org page-level configuration for two main properties:

1.  The product ID that the user must be granted to view the content.
2.  Whether this content requires this product at this time.

The JSON-LD and Microdata formats are supported.

More detail on the markup is available [here](https://developers.google.com/search/docs/data-types/paywalled-content).

#### JSON-LD markup

> Using JSON-LD, the markup would look like:

Reviewer: Will it always look like this, or is this an example?

Author: Updated with "Below is an example of...:"

```html
<script type="application/ld+json">
  {
    "@context": "http://schema.org",
    "@type": "NewsArticle",
    "isAccessibleForFree": false,
    "publisher": {
      "@type": "Organization",
      "name": "The Norcal Tribune"
    },
    "hasPart": {...},
    "isPartOf": {
      "@type": ["CreativeWork", "Product"],
      "name" : "The Norcal Tribune",
      "productID": "norcal_tribune.com:basic"
    }
  }
</script>
```

> Thus, notice that:

Reviewer: Why am I noticing this? Why is it important?

Author: I've merged the two examples as there was repetition.

2.  This document is currently locked (`"isAccessibleForFree": false`).

#### Microdata markup

Using Microdata, the markup could look like this:

```html
<div itemscope itemtype="http://schema.org/NewsArticle">
  <meta itemprop="isAccessibleForFree" content="false" />
  <div
    itemprop="isPartOf"
    itemscope
    itemtype="http://schema.org/CreativeWork http://schema.org/Product">
    <meta itemprop="name" content="The Norcal Tribune" />
    <meta itemprop="productID" content="norcal_tribute.com:basic" />
  </div>
</div>
```

A usable configuration will provide `NewsArticle` typed item with `isAccessibleForFree` property and a subitem of type `Product` that specifies the `productID`.

In this example:

1.  The product ID is "norcal_tribune.com:basic" (`"productID": "norcal_tribune.com:basic"`).
2.  This document is currently locked (`"isAccessibleForFree": false`).

The configuration is resolved as soon as `productID` and `isAccessibleForFree` are found. It is, therefore, advised to place the configuration as high up in the DOM tree as possible.

## Attributes
### `subscription-action`

> An action declared in the "actions" configuration can be marked up using `subscriptions-action` attribute.

Reviewer: Why would I want to mark it up? Whats the value of this attribute?

Author: Added some more text here to describe what the actions do.

Available values:
- `login`: this will trigger the [Login page][4] of the selected service.
- `subscribe`: this will trigger the [Subscribe page][5] of the selected service.

> For instance, this button will execute the "subscribe" action:

Reviewer: ```suggestion
For example, this button will execute the "subscribe" action:
```

```html
<button subscriptions-action="subscribe" subscriptions-display="EXPR">
  Subscribe now
</button>
```

By default, the actions are hidden and must be explicitly shown using the `subscriptions-display` expression.

### `subscription-service`

In the markup the actions can be delegated to other services for them to execute the actions. This can be achieved by specifying `subscriptions-service` attribute.

Available values:
- `local`: this will force the `local` service to be used for a particular action.
- `{serviceId}` (e.g. `subscribe.google.com`): this will force the service with ID `serviceId` to be used for a particular action.

> For instance, this button will the `subscribe.google.com` service to perform the `subscribe` action even when `local` service is selected:

Reviewer: I don't understand this sentence.

Author: Changing to:

> For example, this button will surface the subscribe page from the `subscribe.google.com`  service, regardless of the [service score factors][9]:

Hopefully this makes more sense.

Reviewer: Yes, much better!

Notice, while not explicitly visible, any vendor service can also implement its own actions. Or it can delegate to the `"login"` service to execute `"login"` or `"subscribe"` action.

### Action configuration

The following properties are defined in this configuration:

<table>
  <tr>
    <th>Property</th>
    <th>Values</th>
    <th>Description</th>
  </tr>
  <tr>
    <td class="col-fourty"><code>login</code></td>
    <td class="col-twenty">&lt;URL&gt;</td>
    <td>The HTTPS URL for the <a href="#login-page">Login Page</a>.</td>
  </tr>
  <tr>
    <td class="col-fourty"><code>subscribe</code></td>
    <td class="col-twenty">&lt;URL&gt;</td>
    <td>The HTTPS URL for the <a href="#subscribe-page">Subscribe Page</a>.</td>
  </tr>
</table>

_&lt;URL&gt;_ values specify HTTPS URLs with substitution variables. The substitution variables are covered in more detail in the [URL Variables][13] section above.

Example action configuration:
```js
"actions":{
  "login": "https://pub.com/amp-login?rid=READER_ID&url=SOURCE_URL",
  "subscribe": "https://pub.com/amp-subscribe?rid=READER_ID&url=SOURCE_URL"
}
```

> ### Action Markup

Reviewer: Second words in headings should be lowercase - unless its a proper noun

```html
<template type="amp-mustache" subscriptions-dialog subscriptions-display="NOT granted">
  <!-- Customise the experience for the user using the `data` object returned in the authorization response -->
  <!-- Do NOT use the `data` object to show or hide premium content as this is not always returned -->
  {{^data.articlesRead}}
  <p>
    You have read all of your free articles!
  </p>
  {{/data.articlesRead}}
  {{#data.articlesRead}}
  <p>
    You have read <b>{{data.articlesRead}}</b> articles.
  </p>
  {{/data.articlesRead}}
  {{#data.articlesLeft}}
  <p>
    You have <b>{{data.articlesLeft}}</b> free articles left!
  </p>
  {{/data.articlesLeft}}
  <button subscriptions-action="subscribe" subscriptions-service="local" subscriptions-display="true">
    Subscribe
  </button>
  <section subscriptions-display="NOT granted AND NOT data.isLoggedIn">
  <button
    subscriptions-action="login"
    subscriptions-service="local"
    subscriptions-display="NOT granted AND NOT data.isLoggedIn">
    Already subscribed?
  </button>

</template>
```

The first dialog with matching `subscriptions-display` is shown.

> ## Actions

Reviewer: Could you provide an example of how these actions would be called? I'm unsure of the syntax, but the flow information is great

Author: Done - thanks for all your detailed reviews and comments!

- Triggered when a subscription account linking request initiated by the selected service has been cancelled.
- Data: `serviceId` of the selected service.

> ## AMP Glossary

Reviewer: Please remove this section.
